app.rb

```diff
@@ -0,0 +1,163 @@
+require_relative 'book'
+require_relative 'teacher'
+require_relative 'student'
+require_relative 'rentals'
+
+class App
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+  end
+
+  def create_person
+    print 'Do you want to create a student (1) or a teacher (2)? [Input the number]:'
+    option = gets.chomp
+    case option
+    when '1'
+      create_student
+    when '2'
+      create_teacher
+    else
+      puts 'Invalid option'
+      create_person
+    end
+    puts 'Person created successfully'
+  end
+
+  def create_student
+    print 'Age: '
+    student_age = gets.chomp
+    print 'Name: '
+    student_name = gets.chomp
+    print 'Has parent permission? [Y/N]: '
+    parent_permission = gets.chomp
+    is_permitted = case parent_permission.downcase
+                   when 'y'
+                     true
+                   when 'n'
+                     false
+                   else
+                     puts 'It\'s a yes or no type thing! C\'mon now!'
+                   end
+    student = Student.new(student_age, student_name, is_permitted)
+    @people.push(student)
+  end
+
+  def create_teacher
+    print 'Age: '
+    teacher_age = gets.chomp
+    print 'Name: '
+    teacher_name = gets.chomp
+    print 'Specialization: '
+    teacher_specialization = gets.chomp
+    teacher = Teacher.new(teacher_age, teacher_name, teacher_specialization)
+    @people.push(teacher)
+  end
+
+  def list_people
+    if @people.empty?
+      puts 'No people to display'
+    else
+      @people.each { |person| puts "#{person.class}: Name: #{person.name}, ID: #{person.id}, Age: #{person.age}" }
+    end
+  end
+
+  def create_book
+    print 'Title: '
+    book_title = gets.chomp
+    print 'Author: '
+    book_author = gets.chomp
+    book = Book.new(book_title, book_author)
+    @books.push(book)
+    puts 'Book created successfully'
+  end
+
+  def list_books
+    if @books.empty?
+      puts 'No books to display'
+    else
+      @books.each { |book| puts "Title: #{book.title}, Author: #{book.author}" }
+    end
+  end
+
+  def create_rental
+    if @books.empty? || @people.empty?
+      puts 'There must be at least one book and one person for a rental to take place!'
+    else
+      puts 'Select a book from the following list by number'
+      @books.each_with_index { |book, index| puts "#{index}) Title: \"#{book.title}\", Author: #{book.author}" }
+      book_id = gets.chomp.to_i
+      puts 'Select a person from the following list by number (not ID)'
+      @people.each_with_index do |person, index|
+        puts "#{index}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+      end
+      person_id = gets.chomp.to_i
+      puts 'Enter the date in this format yy/mm/dd: '
+      date = gets.chomp
+      rental = Rental.new(@books[book_id], @people[person_id], date)
+      @rentals.push(rental)
+      puts 'Rental created successfully'
+    end
+  end
+
```

Comment:
- [ ] While trying to create a `rental` I have encountered an error. Please ee the screenshot below and try to fix the error:
<img width="778" alt="Screenshot 2023-10-07 at 9 10 01 PM" src="https://github.com/NalianAchei2020/Shool_Library/assets/44798044/a6f21f24-2276-47f5-9853-05292ac198a6">